packages/express/src/middlewares/authWithJWT.ts

```diff
@@ -16,39 +17,146 @@ function parseAuthorizationHeader(
   return jwt;
 }
 
-export function createAuthWithJWT(publicKey: string, options?: VerifyOptions) {
-  return async function authWithJwt(
-    req: GenericRequest,
-    res: Response,
-    next: NextFunction
-  ): Promise<void> {
-    const logger = getRequestLogger(req);
-    const cookies = (req.cookies as Record<string, string | undefined>) || {};
-    const jwt =
-      cookies.jwt || parseAuthorizationHeader(req.headers.authorization);
-
-    if (!jwt) {
-      logger.debug("auth.authWithJwt.error.notPresent", {
-        headers: req.headers,
-      });
-      res.status(401).json(errorView(new UnauthorizedError(["jwt"])));
-      return;
-    }
+enum AuthMethods {
+  JWT = "JWT",
+  API_KEY = "API_KEY",
+}
 
+interface JWTOptions {
+  jwtVerifyOptions: VerifyOptions;
+  types: [AuthMethods.JWT];
+}
+
+interface TokenOptions {
```

Comment:
este lo escribi mal yo
```suggestion
interface ApiKeyOptions {
```

Reply:
y yo me olvide de cambiarlo